modelzoo/SimpleMultiTask/train.py

```diff
@@ -0,0 +1,781 @@
+
+    task_config = tf_config.get('task')
+    task_type = task_config.get('type')
+    task_index = task_config.get('index') + (1 if task_type == 'worker'
+                                             and chief_hosts else 0)
+
+    if task_type == 'chief':
+        task_type = 'worker'
+
+    is_chief = True if task_index == 0 else False
+    cluster = tf.train.ClusterSpec({'ps': ps_hosts, 'worker': worker_hosts})
+
+    if args.protocol in ['grpc++', 'star_server']:
+        raise ValueError('grpc++ and star_server protocols have not been enabled in the model yet')
+
+    server = tf.distribute.Server(cluster,
+                                  job_name=task_type,
+                                  task_index=task_index,
+                                  protocol=args.protocol)
+    if task_type == 'ps':
+        server.join()
+    elif task_type == 'worker':
+        tf_config = {'ps_hosts': ps_hosts,
+                     'worker_hosts': worker_hosts,
+                     'type': task_type,
+                     'index': task_index,
+                     'is_chief': is_chief}
+
+        tf_device = tf.device(tf.train.replica_device_setter(
+                              worker_device=f'/job:worker/task:{task_index}',
+                              cluster=cluster))
+        return tf_config, server, tf_device
+    else:
+        raise ValueError(f'[TF_CONFIG ERROR] Task type or index error.')
+
+# Some DeepRec's features are enabled by ENV.
+# This func is used to set ENV and enable these features.
+# A triple quotes comment is used to introduce these features and play an emphasizing role.
+def set_env_for_DeepRec():
+    '''
+    Set some ENV for these DeepRec's features enabled by ENV.
+    More Detail information is shown in https://deeprec.readthedocs.io/zh/latest/index.html.
+    START_STATISTIC_STEP & STOP_STATISTIC_STEP: On CPU platform, DeepRec supports memory optimization
+        in both stand-alone and distributed trainging. It's default to open, and the
+        default start and stop steps of collection is 1000 and 1100. Reduce the initial
+        cold start time by the following settings.
+    MALLOC_CONF: On CPU platform, DeepRec can use memory optimization with the jemalloc library.
+        Please preload libjemalloc.so by `LD_PRELOAD=./libjemalloc.so.2 python ...`
+    DNNL_MAX_CPU_ISA: Specify the highest instruction set used by oneDNN (when the version is less than 2.5.0),
+        it will be set to AVX512_CORE_AMX to enable Intel CPU's feature.
+    '''
+    os.environ['START_STATISTIC_STEP'] = '100'
+    os.environ['STOP_STATISTIC_STEP'] = '110'
+    os.environ['MALLOC_CONF'] = \
+        'background_thread:true,metadata_thp:auto,dirty_decay_ms:20000,muzzy_decay_ms:20000'
+
+def check_stock_tf():
+    import pkg_resources
+    detailed_version = pkg_resources.get_distribution('Tensorflow').version
+    return not ('deeprec' in detailed_version)
+
```

Comment:
I tend to remove this automatic detection of the TF version. I think it helps users realize what environment they are using. stock TF or DeepRec.

Reply:
Same comment as in https://github.com/changqi1/DeepRec/pull/63: 
If there won't be any clearer information for failure when a user is using stock TF with unsupported features, first-time users won't have a clue that a problem is a stock version of tf. If automatic detection should be disabled, I think it's still better to have a clear error message when a user would try to use DeepRec features with stock TF.